server/core/basic_cluster.go

```diff
@@ -115,188 +102,289 @@ func (bc *BasicCluster) GetFollowerStores(region *RegionInfo) []*StoreInfo {
 }
 
 // ResetStoreLimit resets the limit for a specific store.
 func (bc *BasicCluster) ResetStoreLimit(storeID uint64, limitType storelimit.Type, ratePerSec ...float64) {
-	bc.Lock()
-	defer bc.Unlock()
+	bc.Stores.mu.Lock()
+	defer bc.Stores.mu.Unlock()
 	bc.Stores.ResetStoreLimit(storeID, limitType, ratePerSec...)
 }
 
 // UpdateStoreStatus updates the information of the store.
-func (bc *BasicCluster) UpdateStoreStatus(storeID uint64, leaderCount int, regionCount int, pendingPeerCount int, leaderSize int64, regionSize int64, witnessCount int) {
-	bc.Lock()
-	defer bc.Unlock()
-	bc.Stores.UpdateStoreStatus(storeID, leaderCount, regionCount, pendingPeerCount, leaderSize, regionSize, witnessCount)
+func (bc *BasicCluster) UpdateStoreStatus(storeID uint64) {
+	bc.Regions.mu.RLock()
+	leaderCount := bc.Regions.GetStoreLeaderCount(storeID)
+	regionCount := bc.Regions.GetStoreRegionCount(storeID)
+	witnessCount := bc.Regions.GetStoreWitnessCount(storeID)
+	pendingPeerCount := bc.Regions.GetStorePendingPeerCount(storeID)
+	leaderRegionSize := bc.Regions.GetStoreLeaderRegionSize(storeID)
+	regionSize := bc.Regions.GetStoreRegionSize(storeID)
+	bc.Regions.mu.RUnlock()
+	bc.Stores.mu.Lock()
+	defer bc.Stores.mu.Unlock()
+	bc.Stores.UpdateStoreStatus(storeID, leaderCount, regionCount, pendingPeerCount, leaderRegionSize, regionSize, witnessCount)
+}
+
+// PutStore put a store.
+func (bc *BasicCluster) PutStore(store *StoreInfo) {
+	bc.Stores.mu.Lock()
+	defer bc.Stores.mu.Unlock()
+	bc.Stores.SetStore(store)
+}
+
+// ResetStores resets the store cache.
+func (bc *BasicCluster) ResetStores() {
+	bc.Stores.mu.Lock()
+	defer bc.Stores.mu.Unlock()
+	bc.Stores.StoresInfo = NewStoresInfo()
+}
+
+// DeleteStore deletes a store.
+func (bc *BasicCluster) DeleteStore(store *StoreInfo) {
+	bc.Stores.mu.Lock()
+	defer bc.Stores.mu.Unlock()
+	bc.Stores.DeleteStore(store)
+}
+
+/* Regions read operations */
+
+// GetRegion searches for a region by ID.
+func (bc *BasicCluster) GetRegion(regionID uint64) *RegionInfo {
+	bc.Regions.mu.RLock()
+	defer bc.Regions.mu.RUnlock()
+	return bc.Regions.GetRegion(regionID)
+}
+
+// GetRegions gets all RegionInfo from regionMap.
+func (bc *BasicCluster) GetRegions() []*RegionInfo {
+	bc.Regions.mu.RLock()
+	defer bc.Regions.mu.RUnlock()
+	return bc.Regions.GetRegions()
+}
+
+// GetMetaRegions gets a set of metapb.Region from regionMap.
+func (bc *BasicCluster) GetMetaRegions() []*metapb.Region {
+	bc.Regions.mu.RLock()
+	defer bc.Regions.mu.RUnlock()
+	return bc.Regions.GetMetaRegions()
+}
+
+// GetStoreRegions gets all RegionInfo with a given storeID.
+func (bc *BasicCluster) GetStoreRegions(storeID uint64) []*RegionInfo {
+	bc.Regions.mu.RLock()
+	defer bc.Regions.mu.RUnlock()
+	return bc.Regions.GetStoreRegions(storeID)
+}
+
+// GetLeaderStoreByRegionID returns the leader store of the given region.
+func (bc *BasicCluster) GetLeaderStoreByRegionID(regionID uint64) *StoreInfo {
+	bc.Regions.mu.RLock()
```

Comment:
Is it better to use the function `GetRegion(regionID uint64)` to decrease the same code?